tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -13,7 +16,33 @@ import (
 // this function requires specifying the credentialAccountID of the user's credential account.
 // The resourceTrackerDir is where the `windows-node-installer.json` file which contains IDs of created instance and
 // security group will be created.
-func CloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, resourceTrackerDir string) (Cloud, error) {
+
+//struct to hold azure specific information.
+type svc struct {
+	resourceGroupName     string // name of the resource group
+	resourceGroupLocation string // name of the resource location
+
+	deploymentName string                 // name of the deployment
+	templateFile   map[string]interface{} // name of the templateFile (need from user)
+	parametersFile map[string]interface{} // name of the parametersFile (need from user)
```

Comment:
Are you saying that the user has to provide these JSON files as input? I am a nack on that and I would like to know why you chose this path. I prefer taking whatever inputs are required as command line parameters. More over not all inputs need to be given by the user as we can apply sensible defaults.

Reply:
You're right. We could approach this two ways, use the exec plugin to execute the az cli commands  or use the azure-sdk-for-go module and supply the arguments to the function that creates an instance. I'm following the second path to create the instance and it is expecting to supply those template and parameters JSON files. But if we can choose the first path then the job would be way simpler.

Reply:
We should chat about this before you proceed.